### DISCUSSION TITLE
Allow disabling internal notebook monitor when external is connected

### DISCUSSION THREAD

**Discussion by thiagoufg** [linuxdeepin/developer-center#2956] [Features Request & Ideas | 特性请求 & 头脑风暴]
Many people need this, and deepin doesn't allow it in an easy way (graphically).

Currently, the internal laptop monitor is always ON, even if there is an external monitor connected and the notebook lid is closed. As a result, the login screen, the dock (launcher), and many other windows are hidden from the users's main monitor (the external), in case he chooses to only use the external.

There are xrandr commands out there, like the one I posted at the end of this issue, but scripts like these are difficult to set up and, in my case, after a reboot, even though the internal notebook monitor was OFF and the external was ON, I had problems with resolution, which were also fixed with some more scripts, and then, I had issues with the task bar (dock, launcher) position, which apparently was considering the resolution of the internal monitor rather than the external.

So, a lot of issues!

By the way, my computer is a SONY VAIO VPCEB and my graphics card is ATI Mobility Radeon HD 5470.

```
Have to create a script in /etc/X11/Xsession.d/ called "45custom_xrandr-settings"

Contents are:

Code:
EXTERNAL_OUTPUT="VGA"
INTERNAL_OUTPUT="LVDS"

xrandr |grep $EXTERNAL_OUTPUT | grep " connected "
if [ $? -eq 0 ]; then
xrandr --output $INTERNAL_OUTPUT --off --output $EXTERNAL_OUTPUT --auto 
else
xrandr --output $INTERNAL_OUTPUT --auto --output $EXTERNAL_OUTPUT --off
fi
At startup this checks if external display is connected.
If it is connected then the laptop monitor turns OFF and external monitor turns ON.
If the external display is NOT connected then only internal monitor works.
```

**Comment by vampywiz17** [comment]
I write a issue at this theme again. it would be a good to fix it... 

https://github.com/linuxdeepin/developer-center/issues/3552

**Comment by meiyixiang** [comment]
Hello, I have recorded your question, and we will consider and evaluate how to achieve it.
